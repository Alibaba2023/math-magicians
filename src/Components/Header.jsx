import './style.css';

function Header() {
  return (
    <div className="container-routers">
      <nav className="header-items">
        <a href="/" className="header-title">Math Magicians</a>
        <div className="navigation-items-box">
          <a href="/" className="navigation-items">Home</a>
          <a href="/calculator" className="navigation-items">Calculator</a>
          <a href="/Quotes" className="navigation-items">Quotes</a>
        </div>
      </nav>
    </div>
  );
}

export default Header;
