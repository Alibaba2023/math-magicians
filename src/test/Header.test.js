import React from 'react';
import { render } from '@testing-library/react';
import Header from '../Components/Header';

describe('Header Component', () => {
  test('renders correctly', () => {
    const { container } = render(<Header />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders the header title correctly', () => {
    const { getByText } = render(<Header />);
    const headerTitle = getByText('Math Magicians');
    expect(headerTitle).toBeInTheDocument();
  });

  test('renders the navigation items correctly', () => {
    const { getByText } = render(<Header />);
    const homeLink = getByText('Home');
    const calculatorLink = getByText('Calculator');
    const quotesLink = getByText('Quotes');
    expect(homeLink).toBeInTheDocument();
    expect(calculatorLink).toBeInTheDocument();
    expect(quotesLink).toBeInTheDocument();
  });
});
