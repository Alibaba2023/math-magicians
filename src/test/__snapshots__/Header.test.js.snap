// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders correctly 1`] = `
<div
  class="container-routers"
>
  <nav
    class="header-items"
  >
    <a
      class="header-title"
      href="/"
    >
      Math Magicians
    </a>
    <div
      class="navigation-items-box"
    >
      <a
        class="navigation-items"
        href="/"
      >
        Home
      </a>
      <a
        class="navigation-items"
        href="/calculator"
      >
        Calculator
      </a>
      <a
        class="navigation-items"
        href="/Quotes"
      >
        Quotes
      </a>
    </div>
  </nav>
</div>
`;
